src/backend/distributed/utils/metadata_cache.c

```diff
@@ -2051,6 +2052,42 @@ CitusTextSendAsJsonbFunctionId(void)
 }
 
 
+/*
+ * PgTableVisibleFuncId returns oid of the pg_table_is_visible function.
+ */
+Oid
+PgTableVisibleFuncId(void)
+{
+	if (MetadataCache.pgTableIsVisibleFuncId == InvalidOid)
+	{
+		const int argCount = 1;
+
+		MetadataCache.pgTableIsVisibleFuncId =
```

Comment:
Could this call fail ?

Reply:
Yes, it does if someone changes the signature of `pg_table_is_visible`. However, that seems like an unlikely case unless a major Postgres version change. 

Also, we're calling `FunctionOid ()` for several other functions as well.

I already added a regression test (e.g., at the beginning of the sql file), which ensures that the signatures are OK.

I cannot think of doing anything if the function returns `InvalidOid` instead of erroring out. We'll need to error out by ourselves?